microsetta_private_api/client_state/redis_cache.py

```
@@ -0,0 +1,31 @@
+import json
+import redis
+
+
+class RedisCache:
+    # System Banner Format: ["Blah blah blah", "primary"]
+    SYSTEM_BANNER = "system_banner"
+
+    def __init__(self):
+        self.r = redis.Redis(host='localhost',
+                             port=6379,
+                             db=0,
+                             decode_responses=True)
+
+    def __getitem__(self, key):
+        val = self.r.get(key)
+        if val is None:
+            return None
+        return json.loads(val)
+
+    def get(self, key, default):
+        val = self.r.get(key)
+        if val is None:
+            return default
+        return json.loads(val)
+
+    def __setitem__(self, key, value):
+        self.r.set(key, json.dumps(value))
+
+    def __delitem__(self, key):
+        self.r.delete(key)
```

what happens if the key is not present?

---

Case with key present:

```
>>> import redis
>>> r = redis.Redis()
>>> r.set("x", "twenty-seven")
True
>>> r.delete("x")
1
```

Case with no key present:
```
>>> import redis
>>> r = redis.Redis()
>>> print(r.delete("No Such Key"))
0
```

Looks like it just silently ignores failing to remove that key.  I'm fine with that as our api, but if you wanted it to be more dict like, we could throw a KeyError

---

I think this is perfect